torchgeo/datasets/spacenet.py

```diff
@@ -191,49 +199,287 @@ def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        * Labels - GeoJSON
+                * label.geojson
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/1807.01232
+
+    .. note::
+
+       This dataset requires the following additional library to be installed:
+
+       * `radiant-mlhub <https://pypi.org/project/radiant-mlhub/>`_ to download the
+         imagery and labels from the Radiant Earth MLHub
+
+    """
+
+    dataset_id = "spacenet2"
+    collection_md5_dict = {
+        "sn2_AOI_2_Vegas": "cdc5df70920adca870a9fd0dfc4cca26",
+        "sn2_AOI_3_Paris": "8299186b7bbfb9a256d515bad1b7f146",
+        "sn2_AOI_4_Shanghai": "4e3e80f2f437faca10ca2e6e6df0ef99",
+        "sn2_AOI_5_Khartoum": "8070ff9050f94cd9f0efe9417205d7c3",
+    }
+
+    imagery = {
+        "MS": "MS.tif",
+        "PAN": "PAN.tif",
+        "PS-MS": "PS-MS.tif",
+        "PS-RGB": "PS-RGB.tif",
+    }
+    label_glob = "label.geojson"
+
+    def __init__(
+        self,
+        root: str,
+        image: str = "PS-RGB",
+        collections: Optional[List[str]] = None,
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        download: bool = False,
+        api_key: Optional[str] = None,
+        checksum: bool = False,
+    ) -> None:
+        """Initialise a new SpaceNet 2 Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            image: image selection which must be in ["MS", "PAN", "PS-MS", "PS-RGB"]
+            collections: collection selection which must be a subset of:
+                         [sn2_AOI_2_Vegas, sn2_AOI_3_Paris, sn2_AOI_4_Shanghai,
+                         sn2_AOI_5_Khartoum]
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory.
+            api_key: a RadiantEarth MLHub API key to use for downloading the dataset
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing
+        """
+        super().__init__(
+            root, image, collections, transforms, download, api_key, checksum
+        )
+
+    # TODO: Remove this once radiantearth/radiant-mlhub#65 is fixed
+    def _load_files(self, root: str) -> List[Dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image and label
+        """
+        files = []
+        for collection in self.collections:
+            images = glob.glob(os.path.join(root, collection, "*", self.filename))
+            images = sorted(images)
+            for imgpath in images:
+                pat = re.compile(r"img1(\\|\/)")
```

Comment:
```suggestion
                pat = re.compile("img1" + re.escape(os.sep))
```

Reply:
Also, this should be moved outside of all the for-loops so we only have to do it once

Reply:
Done